Source/JavaScriptCore/wasm/WasmBBQJIT64.cpp

```diff
@@ -2382,8 +2393,12 @@ PartialResult WARN_UNUSED_RETURN BBQJIT::addI64Xor(Value lhs, Value rhs, Value&
             m_jit.xor64(lhsLocation.asGPR(), rhsLocation.asGPR(), resultLocation.asGPR());
         ),
         BLOCK(
+#if CPU(ARM64)
+            m_jit.xor64(TrustedImm64(ImmHelpers::imm(lhs, rhs).asI64()), ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
+#else
             m_jit.move(ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
             m_jit.xor64(TrustedImm64(ImmHelpers::imm(lhs, rhs).asI64()), resultLocation.asGPR());
+#endif
```

Comment:
Idle thought, but given how this pattern shows up multiple times (esp. since it uses macros) it might be nice to extract this out into some sort of wrapper method which dispatches to either (move-then-arithmetic) or (just-arithmetic) depending on the target.

Reply:
Yeah, we do this elsewhere in the MacroAssembler we should do it here too.

Reply:
Latest push ends up sidestepping this by moving the logic into the macro assembler.